### PR TITLE
ui: toolchain, libraries and siad app, test ci, automatic type generation 

### PR DESCRIPTION
**Pull request by alexfreska**

### CODE REVIEW

File: ui/libs/react-renterd/src/useRhp.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// mux.POST("/api/rhp/scan", srv.rhpScanHandler)
+// mux.POST("/api/rhp/form", srv.rhpFormHandler)
+// mux.POST("/api/rhp/renew", srv.rhpRenewHandler)
+// mux.POST("/api/rhp/read", srv.rhpReadHandler)
+// mux.POST("/api/rhp/append", srv.rhpAppendHandler)
+
+export const foo = 1
```
Comment (lukechampine) [inline ui/libs/react-renterd/src/useRhp.ts:7]: what is this file?

File: ui/migrations.json [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    {
+      "version": "14.1.5-beta.0",
+      "cli": "nx",
+      "description": "Update to export default in jest config and revert jest.preset.ts to jest.preset.js",
+      "factory": "./src/migrations/update-14-1-5/update-exports-jest-config",
+      "package": "@nrwl/jest",
+      "name": "update-to-export-default"
+    }
+  ]
+}
```
Comment (lukechampine) [inline ui/migrations.json:116]: what is this file?

**Comment by lukechampine** [comment]
+40,000 lines of javascript boilerplate and configuration 

 image

File: ui/libs/react-renterd/src/useRhp.ts (hunk 1 above)
Comment (alexfreska) [reply ui/libs/react-renterd/src/useRhp.ts:7]: Its for rhp APIs which aren't implemented yet, I'll add them.

File: ui/migrations.json (hunk 2 above)
Comment (alexfreska) [reply ui/migrations.json:116]: I initially set up the project with nx 13 and migrated it to 14, its an artifact of that and can be removed

**Comment by alexfreska** [comment]
> +40,000 lines of javascript boilerplate and configuration
> 
> image

@lukechampine yes painful 😆, certainly a good bit of boilerplate, a little bit of application code, but almost 100% of the LOC is actually from the `package-lock.json` file lol:
Screen Shot 2022-05-18 at 10 45 23 AM

**Comment by lukechampine** [comment]
what if we kept the UI code in a separate repo? It would complicate the build a little, but I see some compelling advantages: the UI would become optional (e.g. for exchanges, or power users, or people who want maximum security), and you'd be free to develop the UI rapidly without cluttering backend development.

**Pull request closed without merging**